installscripts/jazz-terraform-unix-noinstances/scripts/gitlab-trigger-job.sh

```diff
@@ -0,0 +1,159 @@
+          <triggerOnClosedMergeRequest>false</triggerOnClosedMergeRequest>
+          <triggerOpenMergeRequestOnPush>never</triggerOpenMergeRequestOnPush>
+          <triggerOnNoteRequest>true</triggerOnNoteRequest>
+          <noteRegex>Jenkins please retry a build</noteRegex>
+          <ciSkip>true</ciSkip>
+          <skipWorkInProgressMergeRequest>true</skipWorkInProgressMergeRequest>
+          <setBuildDescription>true</setBuildDescription>
+          <branchFilterType>All</branchFilterType>
+          <includeBranchesSpec></includeBranchesSpec>
+          <excludeBranchesSpec></excludeBranchesSpec>
+          <targetBranchRegex></targetBranchRegex>
+          <secretToken>replace</secretToken>
+        </com.dabsquared.gitlabjenkins.GitLabPushTrigger>
+      </triggers>
+    </org.jenkinsci.plugins.workflow.job.properties.PipelineTriggersJobProperty>
+  </properties>
+  <definition class="org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition" plugin="workflow-cps@2.36">
+   <script>#!groovy.
+
+node  {
+//echo sh(returnStdout: true, script: &apos;env&apos;)
+
+def gitlab_credentialsId   = env.REPO_CREDENTIAL_ID
+def gitlab_repo_base       = env.REPO_BASE
+def gitlab_repo_url        = &quot;http://&quot; + gitlab_repo_base + &quot;/&quot; + env.REPO_LOC + &quot;/&quot;
+
+def jenkins_username  = env.JENKINS_USERNAME
+def jenkins_password  = env.JENKINS_PASSWORD
+
+def gitlab_repo_name  = env.gitlabSourceRepoName
+def gitlab_branch     = env.gitlabBranch
+def service_type = &apos;&apos;
+def service_name = &apos;&apos;
+def domain_name  = &apos;&apos;
+
+echo &quot;gitlab_credentialsId:gitlab_credentialsId&quot;
+echo &quot;gitlab_repo_url:$gitlab_repo_url&quot;
+echo &quot;gitlab_repo_name:$gitlab_repo_name&quot;
+echo &quot;gitlab_branch:$gitlab_branch&quot;
+
+try{
+       sh &apos;rm -rf $gitlab_repo_name*&apos;
+       dir(gitlab_repo_name)
+       {
+               checkout([$class: &apos;GitSCM&apos;, branches: [[name: &apos;*/master&apos;]], doGenerateSubmoduleConfigurations: false, extensions: [], submoduleCfg: [], userRemoteConfigs: [[credentialsId: gitlab_credentialsId, url: gitlab_repo_url + gitlab_repo_name + &apos;.git&apos;]]])
+
+          //Find the service type
+          if (fileExists(&apos;build.api&apos;)) {
+                               service_type = &quot;api&quot;
+          } else if(fileExists(&apos;build.lambda&apos;)) {
+                               service_type = &quot;lambda&quot;
+          } else if(fileExists(&apos;build.website&apos;)) {
+                               service_type = &quot;website&quot;
+          }
+
+          //Read the JenkinsFile to find the service name and domain
+           def jenkinsfile_dir = sh (
+                      script: &quot;pwd&quot; ,
+                      returnStdout: true
+              ).trim()
+          myFileName = jenkinsfile_dir + &quot;/Jenkinsfile&quot;
+
+          //Read the Jenkins File
+          def curl_cmd = &apos;&apos;
```

Comment:
There were multiple bugs with this code that we had fixed as part of a PR in core for gitlabs. Now the same bugs are showing up here, so please review it and let me know if this required here.

Reply:
addressed.